### ISSUE TITLE
[feature] add missing compiler version check for ti-cgt-arm

### ISSUE THREAD

**Issue by pasrom** [conan-io/conan#7480]
I'm using a cmake toolchain file to compile static libraries with the [ti-cgt-arm](https://www.ti.com/tool/download/ARM-CGT-18) compiler. I have modified the `settings.yml` compiler section with following information:
```
compiler:
    ti-arm-cgt:
        version: ["18.12", "20.2"]
```

For now I'm getting this warning:

```
-- The C compiler identification is TI 18.12.5
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan: Adjusting fPIC flag (ON)
-- This project seems to be plain C, using 'TI' compiler
-- WARN: Unknown compiler 'ti-arm-cgt', skipping the version check...
```

I'm expecting from conan, that the ti-arm-cgt compiler version check is working. My suggestion is to add the version check for ti-cgt-arm to somewhere [here](https://github.com/conan-io/conan/blob/7329ee917f48972c60e93cdf4f69b4c2c42ffcde/conans/client/generators/cmake_common.py#L527-L531), e.g.:
```
elseif(CONAN_COMPILER STREQUAL "ti-cgt-arm")
    conan_split_version(${CONAN_COMPILER_VERSION} CONAN_COMPILER_MAJOR CONAN_COMPILER_MINOR)
    if(NOT ${VERSION_MAJOR}.${VERSION_MINOR} VERSION_EQUAL ${CONAN_COMPILER_MAJOR}.${CONAN_COMPILER_MINOR})
        conan_error_compiler_version()
    endif()
```

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @pasrom 

I am afraid we cannot merge code like ``if(CONAN_COMPILER STREQUAL "ti-cgt-arm")`` that references some custom compiler that the codebase is not aware of, as it would be inconsistent and confusing.

In any case, we are moving away of these compiler checks provided by the ``cmake`` generator, and the current way of thinking is:

- ``generators`` will only provide information about the dependencies. Nothing else, nothing related to the current package build, and not any check regarding the current settings. Only about dependencies.
- ``toochains`` will contain the information about the current build. This would be the place to include compiler checks. They will be more extensible too , and they might be using the new ``templates`` extension. Have a look at:
  - https://docs.conan.io/en/latest/creating_packages/toolchains.html
  - https://docs.conan.io/en/latest/extending/template_system.html

So not there yet, but at least this is the plan, and I think it will be good to support custom compiler checks integrated in the Conan builds.

**Comment by memsharded** [comment]
Hi again @pasrom 

I see you closed #7481, thanks for understanding. I think adding the suggested functionality in your own CMakeLists.txt can be done, do not hesitate to ask for guidance if necessary.

I am going to close this issue, we will definitely take into account the feedback when we move the compiler checks to the ``toolchain()`` feature. Thanks!
